src/sql_types.rs

```diff
@@ -21,6 +22,7 @@ pub struct ColumnPermissions {
 pub struct ColumnDirectives {
     pub name: Option<String>,
     pub description: Option<String>,
+    pub mappings: Option<BiBTreeMap<String, String>>,
```

Comment:
which allows this to be removed

Reply:
My main concern here is that this way, `Column` would need to contain a reference to Schema and we'd do a (slower) traversal _each time_ we prepare the query.

Alternatively, this mapping information can be added to `Column` after the context is loaded by traversing the schema, but the outcome of this would be virtually indistinguishable from loading it from the SQL, and it still adds a one-time penalty for the resolution.

To me, it looked like a bit of duplication in the contexts is essentially faster and requires no significant type changes. Unless I am missing something?

Reply:
Yeah that is an issue but duplication in the SQL context has been a headache in the past because it's hard to track down everywhere that an update needs to be made.

For example, when user defined functions are introduced, we'll need the enum mappings for that too and would have to re-duplicate.

How about wrapping

```rust
pub struct ColumnBuilder {
  ...
  pub column: Arc<Column>
}
```
with an rust enum such that this is always `Arc<Column>, Arc<Type>` or `Arc<Column>, Arc<Enum>`?

---
During schema introspection and parsing user input the `Enum` is available so those are less of an issue

Reply:
It feels wrong to do this for ColumnBuilder as it is not very different from the SQL context population: sticking in data where it is used, not at the SQL query time but upon context loading. What if we need to have type information somewhere else? Should we augment those types as well? It'll get messy quickly, and your concern about duplication in the SQL context will hold just one level above.

To address this problem properly, we may need to review how context is loaded. For example, what if instead of storing `type_oid` as `u32` across the code base and if we refactor `Type`/`TypeCategory` to contain a reference to the actual type (`Arc<Enum>`, `Arc<Composite>`, etc.), then at any point, we can look up the type.

However, this is a much bigger change than the intended scope of the problem in this PR. What if we put a `TODO` kind of note in the SQL context concerning the duplication and the link to this discussion, get this merged (so that the users can benefit from the fix) and then work on a bigger refactoring?  Happy to discuss its details.

Reply:
I've added TODOs with links to this discussion.